components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RDBMSSessionDataStore.java

```diff
@@ -346,25 +247,62 @@ public void persistSessionData(String key, String type, Object entry, long nanoT
             return;
         }
         PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+
         try {
-            String sqlQuery = getSessionStoreDBQuery(sqlInsertSTORE, type);
-            preparedStatement = connection.prepareStatement(sqlQuery);
+            if (StringUtils.isBlank(sqlSelect)) {
+                if (connection.getMetaData().getDriverName().contains(MYSQL_DATABASE)
+                        || connection.getMetaData().getDriverName().contains(H2_DATABASE)) {
+                    sqlSelect = SQL_DESERIALIZE_OBJECT_MYSQL;
+                } else if (connection.getMetaData().getDatabaseProductName().contains(DB2_DATABASE)) {
+                    sqlSelect = SQL_DESERIALIZE_OBJECT_DB2SQL;
+                } else if (connection.getMetaData().getDriverName().contains(MS_SQL_DATABASE)
+                        || connection.getMetaData().getDriverName().contains(MICROSOFT_DATABASE)) {
+                    sqlSelect = SQL_DESERIALIZE_OBJECT_MSSQL;
+                } else if (connection.getMetaData().getDriverName().contains(POSTGRESQL_DATABASE)) {
+                    sqlSelect = SQL_DESERIALIZE_OBJECT_POSTGRESQL;
+                } else if (connection.getMetaData().getDriverName().contains(INFORMIX_DATABASE)) {
```

Comment:
Since `connection.getMetaData().getDriverName()` repeats multiple times, we can use a private method to check DB type... Ex: `getDBType`